data/transactions/logic/eval.go

```diff
@@ -291,7 +423,12 @@ func Check(program []byte, params EvalParams) (cost int, err error) {
 	cx.EvalParams = params
 	cx.program = program
 	for (cx.err == nil) && (cx.pc < len(cx.program)) {
+		prevpc := cx.pc
 		cost += cx.checkStep()
+		if cx.pc == prevpc {
```

Comment:
this is a good check! you might consider making it `<=` to catch even more potential problems